### DISCUSSION TITLE
:class incorrectly applied when using wire:navigate and browser back button

### DISCUSSION THREAD

**Discussion by bezanis** [livewire/livewire#6686] [7. Bugs (archived)] [answered]
### Livewire version

3.0.1

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

While using Livewire 3.0.1 with the included version of Alpine along with Tailwind, I'm experiencing an issue with a class being applied after wire:navigate followed by a browser back button press.
To replicate:
Add a div with a x-data param.
Inside that, another div has a :class with a text class added conditionally based on the value in the x-data. If I navigate to another page normally, the :class logic gets reapplied correctly when I use the browser back to return to the page.
However, if I use wire:navigate to leave the page, and then use the browser back button, the conditional logic is incorrectly still applied to the previous state, along with the default state.

Code:

```
<div  x-data="{ locations:'500' }">
    <div class="text-gray">
        <div @click="locations='1'" :class="(locations === '1') && 'text-orange'">1 Location</div>
        <div @click="locations='500'" :class="(locations === '500') && 'text-orange'">500 Locations</div>
    </div>
</div>
```
In this situation, locations is initialized to 500. The 1 Location text starts as gray and the 500 Locations text starts as orange.
I click '1 Location' which turns 1 Location orange and 500 Locations gray. If I then use a link on the page with wire:navigate to leave the page, and then use the browser back button, both 1 Location and 500 Locations texts are orange. However, if I instead navigate away from the page using a normal link, then return, the colors are reset correctly.

### Additional information

Also reported in alpine repo but closed
https://github.com/alpinejs/alpine/discussions/3749

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by danie-ramdhani** [comment]
yeah, it seems when we click the back button, some alpine component is not working properly.

but, this should work for your case:

```blade
    <div x-data="{ locations: '500' }">
        <div class="text-gray-400">
            <div @click="locations='1'" :class="{
                'text-orange-500': locations === '1'
            }">
                1 Location
            </div>
            <div @click="locations='500'"
                 :class="{
                     'text-orange-500': locations === '500'
                 }">
                500 Location
            </div>
        </div>
    </div>
```

**Reply by bezanis** [reply]
Thanks, this workaround allows me to accomplish the same goal. I'll keep this bug up since the other syntax still causes issues

**Reply by danie-ramdhani** [reply, marked as answer]
Made a PR https://github.com/livewire/livewire/pull/8154

**Reply by danie-ramdhani** [reply]
PR is merged with fix bug, should be fixed in next release.
